src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -1217,6 +1219,149 @@ export default class CanvasController {
         cc.restore();
     }
 
+    /**
+     * Draw the `MeasureTool` path and text labels
+     *
+     * @for CanvasController
+     * @method _drawMeasureTool
+     * @param cc {HTMLCanvasContext}
+     * @private
+     */
+    _drawMeasureTool(cc) {
+        if (!MeasureTool.hasPaths) {
+            return;
+        }
+
+        const pathInfoList = MeasureTool.buildPathInfo();
+
+        cc.save();
+        cc.translate(CanvasStageModel.halfWidth, CanvasStageModel.halfHeight);
+
+        pathInfoList.forEach((pathInfo) => {
+            this._drawMeasureToolPath(cc, pathInfo);
+            this._drawMeasureToolLabels(cc, pathInfo);
+        });
+
+        cc.restore();
+    }
+
+    /**
+     * Draw the `MeasureTool` text labels
+     *
+     * @for CanvasController
+     * @method _drawMeasureToolLabels
+     * @param cc {HTMLCanvasContext}
+     * @param pathInfo {object} as returned by `MeasureTools.getPointsAndLabels()`
+     * @private
+     */
+    _drawMeasureToolLabels(cc, pathInfo) {
+        let leg = pathInfo.firstLeg.next;
+        const values = [];
+
+        // This way the points are translated only once
+        while (leg != null) {
+            // Ignore empty labels
+            if (leg.labels !== null && leg.labels.length !== 0) {
+                const position = CanvasStageModel.translatePostionModelToPreciseCanvasPosition(leg.midPoint);
+
+                values.push({
+                    x: position.x,
+                    y: position.y,
+                    labels: leg.labels
+                });
+            }
+
+            leg = leg.next;
+        }
+
+        // TODO: Create its own color in theme
+        // Label backgrounds
+        cc.fillStyle = this.theme.SCOPE.RUNWAY_EXTENDED_CENTERLINE;
+
+        values.forEach((item) => {
+            const { x, y, labels } = item;
+            const height = 10 + (12 * labels.length);
+            const width = 150;
+
+            cc.fillRect(x, y, width, height);
+        });
```

Comment:
Want to try to get the text box fill to always match the size of its content so it isn't taking up extra space that it doesn't need to (and thus blocking things behind it).

![image](https://user-images.githubusercontent.com/5103735/68353300-252e9580-00d7-11ea-99fc-b49838b98760.png)

Reply:
I actually moved the colours to the themes, but stupidly forgot to push... 🤦‍♂️

I'll merge your changes and do the measurements (I actually removed the text box background for the default theme to march the style of the aircraft labels).

I also slightly changed the behaviour of start measuring to throw off it's called a second time so it can be tested for.